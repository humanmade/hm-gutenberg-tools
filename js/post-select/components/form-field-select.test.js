// import jest from 'jest';
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import FormFieldSelect from './form-field-select';

const options = [
	{
		value: 1,
		label: 'Option 1',
	},
	{
		value: 2,
		label: 'Option 2',
	},
];

test( 'Form Search Field', () => {
	const onChangeHandler = jest.fn();
	const renderer = new ShallowRenderer();

	renderer.render(
		<FormFieldSelect
			fieldId="test-field"
			label="Test Select Field Label"
			onChange={ onChangeHandler }
			options={ options }
			isLoading={ false }
			onFetchMoreTerms={ jest.fn() }
			onUpdateSearch={ jest.fn() }
		/>
	);

	const result = renderer.getRenderOutput()
	expect( result ).toMatchSnapshot();

	// Test that onChange is called correctly.
	result.props.children.props.onChange( [ { value: 'a' } ] );
	expect( onChangeHandler.mock.calls.length ).toBe( 1 );
	expect( onChangeHandler.mock.calls[0][0] ).toHaveLength( 1 );
	expect( onChangeHandler.mock.calls[0][0][0] ).toBe( 'a' );
} );
