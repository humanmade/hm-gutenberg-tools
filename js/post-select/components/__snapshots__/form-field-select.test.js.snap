// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Search Field 1`] = `
<FormRow
  label="Test Select Field Label"
  labelFor="test-field"
>
  <StateManager
    defaultInputValue=""
    defaultMenuIsOpen={false}
    defaultValue={null}
    id="test-field"
    isLoading={false}
    isMulti={true}
    maxMenuHeight={300}
    onChange={[Function]}
    onInputChange={[Function]}
    onMenuScrollToBottom={[Function]}
    options={
      Array [
        Object {
          "label": "Option 1",
          "value": 1,
        },
        Object {
          "label": "Option 2",
          "value": 2,
        },
      ]
    }
  />
</FormRow>
`;
